FWCore/Reflection/test/run_checkClassVersionUpdate.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash -e
+
+SCRAM_TEST_NAME=TestFWCoreReflectionCheckClassVersionUpdate
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+
+# Create a new CMSSW dev area and build modified DataFormats/TestObjects in it
+NEW_CMSSW_BASE=$(/bin/pwd -P)/$CMSSW_VERSION
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+pushd $CMSSW_VERSION/src
+
+# Copy FWCore/Reflection code to be able to edit it to make ROOT header parsing to fail
+for DIR in ${CMSSW_BASE} ${CMSSW_RELEASE_BASE} ${CMSSW_FULL_RELEASE_BASE} ; do
+    if [ -d ${DIR}/src/FWCore/Reflection ]; then
+        mkdir FWCore
+        cp -Lr ${DIR}/src/FWCore/Reflection FWCore/
+        break
+    fi
+done
+if [ ! -e FWCore/Reflection ]; then
+    echo "Failed to symlink FWCore/Reflection from local or release area"
+    exit 1;
+fi
+
+# Modify the IntObject class to trigger a new version
+#
+# Just setting USER_CXXFLAGS for scram is not sufficient,because
+# somehow ROOT (as used by edmCheckClassVersion) still picks up the
+# version 3 of the class
+echo "#define FWCORE_REFLECTION_TEST_INTOBJECT_V4" | cat - FWCore/Reflection/test/stubs/TestObjects.h > TestObjects.h.tmp
+mv TestObjects.h.tmp FWCore/Reflection/test/stubs/TestObjects.h
```

Comment:
@makortel , during PR tests , bot removes the write permissions of all sub-directories under `src` tree (https://github.com/cms-sw/cms-bot/blob/master/pr_testing/run-pr-qa-unit.sh#L104 ). In this case you are copying 
`FWCore/Reflection` from `CMSSW_BASE` ( which does not have write permissions). I would suggest to add `chmod +w FWCore/Reflection/test/stubs` before the move command

Reply:
Ah, thanks! That explains the test failure (about permissions)